src/workspace-data/useDataTableProvider.ts

```diff
@@ -33,98 +37,122 @@ export const useDataTableProvider = (
+            break;
+          case appStatuses.running.status:
+            setWdsApp({ status: 'Ready', state: foundApp });
+            setWdsUrl({ status: 'Ready', state: foundApp.proxyUrls?.wds });
+            break;
+          case appStatuses.error.status:
+            setWdsApp({ status: 'Error', state: foundApp, error: 'Error in WDS app' }); // todo get real error?
+            break;
+          default:
+            if (foundApp?.status) {
+              // eslint-disable-next-line no-console
+              console.log(`Unhandled state [${foundApp?.status} while polling WDS`);
+            }
+        }
+      })
+      .catch((error) => {
+        setWdsApp({ status: 'Error', state: null, error: 'Error resolving WDS app' });
+        reportError('Error resolving WDS app', error);
+      });
+  }, [setWdsApp, workspaceId]);
 
-  const loadWdsTypes = useCallback(
-    (url, workspaceId) => {
-      setWdsTypes({ status: 'None' });
+  const loadWdsTypes = useCallback(() => {
+    // setWdsTypes({ status: 'None' }); //TODO I don't think we need this since we initialize it to None
+    if (wdsUrl.status === 'Ready') {
       return Ajax(signal)
-        .WorkspaceData.getSchema(url, workspaceId)
+        .WorkspaceData.getSchema(wdsUrl.state, workspaceId)
         .then((typesResult) => {
           setWdsTypes({ status: 'Ready', state: typesResult });
         })
         .catch((error) => {
           setWdsTypes({ status: 'Error', state: null, error: 'Error loading WDS schema' });
           reportError('Error loading WDS schema', error);
         });
-    },
-    [signal, setWdsTypes]
-  );
+    }
+  }, [signal, setWdsTypes, wdsUrl, workspaceId]);
 
-  const loadWdsCapabilities = useCallback(
-    async (url) => {
+  const loadWdsCapabilities = useCallback(async () => {
+    if (wdsUrl.status === 'Ready') {
       try {
-        const capabilitiesResult = await Ajax(signal).WorkspaceData.getCapabilities(url);
+        const capabilitiesResult = await Ajax(signal).WorkspaceData.getCapabilities(wdsUrl.state);
         setWdsCapabilities({ status: 'Ready', state: capabilitiesResult });
       } catch (error) {
         setWdsCapabilities({ status: 'Error', state: null, error: 'Error loading WDS capabilities' });
         reportError('Error loading WDS capabilities', error);
       }
-    },
-    [signal, setWdsCapabilities]
-  );
+    }
+  }, [signal, setWdsCapabilities, wdsUrl]);
 
   const loadWdsData = useCallback(async () => {
-    // Try to load the proxy URL
-    if (!wdsApp || !['Ready', 'Error'].includes(wdsApp.status)) {
-      const foundApp = await loadWdsApp(workspaceId);
-      // TODO: figure out how not to make this redundant fetch, per
-      // https://github.com/DataBiosphere/terra-ui/pull/4202#discussion_r1319145491
-      if (foundApp) {
-        loadWdsTypes(foundApp.proxyUrls?.wds, workspaceId);
-        loadWdsCapabilities(foundApp.proxyUrls?.wds);
+    if (useCwds.status === 'Ready') {
```

Comment:
This method will be expanded to have different behavior depending on the value of `useCwds`; in addition it's passed back to the parent component to be used to refresh when polling for a loading WDS app.